File: go-vllm-api/internal/model/image.go
package model

const (
	MaxImageSizeMB   = 10
	MaxImageSizeBytes = MaxImageSizeMB * 1024 * 1024
	MaxWidth         = 8192
	MaxHeight        = 8192
)

var SupportedFormats = map[string]bool{
	"png": true, "jpg": true, "jpeg": true, "gif": true,
	"webp": true, "bmp": true, "tiff": true,
}

type ImageGenerationRequest struct {
	Model           string `json:"model"`
	Prompt          string `json:"prompt"`
	N               int    `json:"n,omitempty"`
	Size            string `json:"size,omitempty"`
	ResponseFormat  string `json:"response_format,omitempty"`
}

type ImageGenerationResponse struct {
	Created int64        `json:"created"`
	Data    []ImageData  `json:"data"`
}

type ImageData struct {
	URL           string `json:"url,omitempty"`
	B64JSON       string `json:"b64_json,omitempty"`
	RevisedPrompt string `json:"revised_prompt,omitempty"`
}

type ImageValidationRequest struct {
	ImageData string `json:"image_data"`
}

type ImageUploadResponse struct {
	Success   bool        `json:"success"`
	Message   string      `json:"message"`
	ImageInfo interface{} `json:"image_info,omitempty"`
}

File: app-controller/schemas/image.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any

from .chat import (
    MAX_IMAGE_SIZE_MB, MAX_WIDTH, MAX_HEIGHT, SUPPORTED_FORMATS,
    validate_image_data, decode_base64_image
)


class ImageGenerationRequest(BaseModel):
    model: str = Field(default="dall-e-3")
    prompt: str = Field(min_length=1, max_length=4000)
    n: Optional[int] = Field(default=1, ge=1, le=10)
    size: Optional[str] = Field(default="1024x1024", pattern=r'^\d+x\d+$')
    quality: Optional[str] = Field(default="standard", pattern=r'^(standard|hd)$')
    style: Optional[str] = Field(default="vivid", pattern=r'^(vivid|natural)$')
    response_format: Optional[str] = Field(default="url", pattern=r'^(url|b64_json)$')


class ImageData(BaseModel):
    b64_json: Optional[str] = None
    url: Optional[str] = None
    revised_prompt: Optional[str] = None


class ImageGenerationResponse(BaseModel):
    created: int
    data: List[ImageData]


class ImageValidationRequest(BaseModel):
    image_data: str


class ImageUploadResponse(BaseModel):
    success: bool
    message: str
    image_info: Optional[Dict[str, Any]] = None
